### PR TITLE
feat: Added support for fs.glob in Node 22+

### LINKED ISSUE #2336: Add instrumentation for `fs.glob` for node 22+
### Description
In #2304 I identified a useful method to add to our [fs instrumentation](https://github.com/newrelic/node-newrelic/blob/f8e2e8efab4fe01b7a5650361578c0f1a5dd2e7b/lib/instrumentation/core/fs.js#L15).  We purposely avoid instrumenting the sync methods because they are used in tight loops in other parts of core and frameworks so you’d get a pile of overhead and thousands of segment you don’t want (could potentially merge them but overhead maybe not worth it)

### Acceptance Criteria
Add instrumentation for `fs.glob` in Node 22+

### Design Consideration/Limitations
_Why is this the route we should take to achieve our goal?_
_What can't be achieved within this story?_

### Dependencies
_Do any other teams or parts of the New Relic product need to be considered?_
_Some common areas: UI, collector, documentation_

### Additional context
We need to add an integration test for this new method.  Looking at the [tests](https://github.com/newrelic/node-newrelic/blob/f8e2e8efab4fe01b7a5650361578c0f1a5dd2e7b/test/integration/core/fs.tap.js) we could reduce a lot of boiler plate too, but up to the dev that takes this on

### PR DESCRIPTION
**Pull request by jsumners-nr**
This PR resolved #2336.

### CODE REVIEW

File: test/unit/instrumentation/core/fs.test.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  t.context.agent = helper.instrumentMockedAgent()
+  t.context.fs = require('node:fs')
+})
+
+tap.afterEach((t) => {
+  helper.unloadAgent(t.context.agent)
+  removeModules(['node:fs'])
+})
+
+tap.test('stat method gets instrumented', (t) => {
```
Comment (bizob2828) [inline test/unit/instrumentation/core/fs.test.js:27]: we already have tests in `tests/integration/fs.tap.js`. I would add the glob tests there and remove this

File: test/unit/instrumentation/core/fs.test.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      const segment = tx.trace.root.children.find((child) => child.name === 'Filesystem/stat')
+      t.ok(segment, 'has stat segment')
+
+      tx.end()
+      t.end()
+    })
+  })
+})
+
+tap.test('glob method gets instrumented', { skip: nodeMajor < 22 }, (t) => {
```
Comment (bizob2828) [inline test/unit/instrumentation/core/fs.test.js:47]: when migrating this test over, the assertions are with segments and metrics

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: test/integration/core/fs.tap.js [hunk 3]
```diff
@@ -13,6 +13,8 @@ const helper = require('../../lib/agent_helper')
 const verifySegments = require('./verify')
 const NAMES = require('../../../lib/metrics/names')
 
+const nodeMajor = parseInt(process.version.split('.')[0].replace('v', ''), 10)
```
Comment (bizob2828) [inline test/integration/core/fs.tap.js:16]: this is fine but you could just use `semver` package.  

```js
const isGlobSupported = semver.satisfies(process.version, '>=22.0.0')
```

**Review by bizob2828 — DISMISSED** [review]

File: test/integration/core/fs.tap.js (hunk 3 above)
Comment (jsumners-nr) [reply test/integration/core/fs.tap.js:16]: We don't have that in our dev dependencies and it seems a bit excessive for this easy check.

Comment (bizob2828) [reply test/integration/core/fs.tap.js:16]: we totally have semver it's all over but i understand if you don't want to update

**Review by bizob2828 — DISMISSED** [review]

File: test/integration/core/fs.tap.js (hunk 3 above)
Comment (jsumners-nr) [reply test/integration/core/fs.tap.js:16]: Ah! It's in the regular dependencies list. I'll update.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**